doc/contributing/building_ruby.md

```diff
@@ -157,6 +157,14 @@ with the Ruby script you'd like to run. You can use the following make targets:
 * `make lldb-ruby`: Runs `test.rb` using Ruby in lldb
 * `make gdb-ruby`: Runs `test.rb` using Ruby in gdb
 
+### Compiling for Debugging
+
+You should configure Ruby without optimization and other flags that may interfere with debugging:
+
+``` shell
+./configure --enable-debug-env cppflags="-fno-omit-frame-pointer" optflags=-O0 LDFLAGS="-fno-omit-frame-pointer"
```

Comment:
`-fno-omit-frame-pointer` isn't a flag for `cpp`.
And `$(optflags)` is included in `LDFLAGS` not only in `CFLAGS`.

```suggestion
./configure --enable-debug-env optflags="-O0 -fno-omit-frame-pointer"
```

`--enable-devel` (which implies `debug-env`) may be worth to mention too.

Reply:
Ah, yes, you are totally correct.